app/services/nomis/client.rb

```diff
@@ -64,5 +87,20 @@ def params_options(method, params)
         #   }
       end
     end
+
+    def auth_header
+      return unless @client_token && @client_key
+
+      token = auth_token(@client_token, @client_key)
```

Comment:
Since the ```@client_token``` and ```@client_key``` are already set, there is need to pass them as arguments to ```#auth_token``` they can be directly accessed from within the ```#auth_token``` method.